### PR TITLE
Pass request agent to cloudant-follow

### PR DESCRIPTION
**Pull request by smithsz**
Pass request agent to cloudant-follow.

See related PR https://github.com/cloudant-labs/cloudant-follow/pull/26.

### CODE REVIEW

**Review by tomblench — APPROVED** [review]

File: tests/helpers/unit.js [hunk 1]
```diff
@@ -90,6 +90,10 @@ helpers.unit = function(method, error) {
           stub.db = helpers.couch + stub.db;
         }
 
+        if (method[1] === 'follow' || method[1] === 'followUpdates') {
+          delete req.httpAgent; // ignore 'httpAgent' in deep equal assert
```
Comment (ricellis) [inline tests/helpers/unit.js:94]: So if the `httpAgent` is getting passed through from `nano` to follow shouldn't it always be the same `httpAgent`?
Or do the tests here use multiple nano instances to assert?

**Review by ricellis — COMMENTED** [review]
Just 1 (2 part) question

File: tests/helpers/unit.js (hunk 1 above)
Comment (ricellis) [reply tests/helpers/unit.js:94]: As per discussion I see the assertion is just against a mock `{db: ...}` object so this makes sense.

**Review by ricellis — APPROVED** [review]
+1

**Pull request merged**